### PR TITLE
Send order/uniswap costs to http solver

### PR DESCRIPTION
**Pull request by marcovc**
I've left a few questions in the code.

### CODE REVIEW

**Review by vkgnosis — COMMENTED** [review]
I feel that the gas price should be part of the Solver trait solver function because it's always relevant information (even if other solvers don't use it) and it means we don't have to make decisions here like what parameters to give the gas estimation or what to do if estimation is impossible. What do you think?

File: solver/src/http_solver.rs [hunk 1]
```diff
@@ -23,9 +24,11 @@ use std::{
     sync::Arc,
 };
 
+const GAS_PER_ORDER: u128 = 37676;
```
Comment (fleupold) [inline solver/src/http_solver.rs:27]: Maybe add a comment on how you derived at the values. I feel they are a bit low, given that a settlement with one trade and Uniswap already costs > 180k gas.

File: solver/src/http_solver.rs [hunk 2]
```diff
@@ -272,6 +289,22 @@ impl HttpSolver {
         serde_json::from_str(text.as_str())
             .with_context(|| format!("failed to decode response json, {}", context()))
     }
+
+    async fn order_cost(&self) -> Result<u128> {
+        // Q: should we use estimate_with_limits below instead?
+        // Q: are the units (gwei) correct?
+        let gas_price_gwei = self.gas_price_estimator.estimate().await? * 1e9;
```
Comment (fleupold) [inline solver/src/http_solver.rs:296]: I think the price estimator returns the price in WEI so no need to multiply with 1e9

**Review by fleupold — COMMENTED** [review]
Looks good! I like @vkgnosis suggestion of passing in the current gas price in the `solve(&self, orders: Vec<Liquidity>)` function as it seems relevant to a lot of solvers. 

The gas price is also be relevant when we evaluate the objective function in the end (it doesn't yet, but should eventually also take execution cost into account)

File: solver/src/driver.rs [hunk 3]
```diff
@@ -162,14 +162,21 @@ impl Driver {
 
         self.metrics.liquidity_fetched(&liquidity);
 
+        let gas_price = self
+            .gas_price_estimator
+            .estimate()
+            .await
+            .context("failed to estimate gas price")?;
+        tracing::debug!("solving with gas price of {}", gas_price);
```
Comment (marcovc) [inline solver/src/driver.rs:170]: Should we use a default price instead?

File: solver/src/http_solver.rs (hunk 1 above)
Comment (marcovc) [reply solver/src/http_solver.rs]: Done. For "better" estimations we should probably use a table close to the values we are expecting (I guess we aren't expecting 50 trades).

File: solver/src/driver.rs (hunk 3 above)
Comment (vkgnosis) [reply solver/src/driver.rs:170]: I think it's correct to fail in that case. We definitely need the gas price estimation to work for submitting our transactions.

**Review by vkgnosis — APPROVED** [review]

**Pull request merged**